### PR TITLE
Implement CRUD on cake feature.

### PR DESCRIPTION
**Pull request by kizzanaome**
### What does this PR do?
 Performs a CRUD on the cake feature

### Description of Task to be completed?
  A user or client can create, read, update, and delete a cake.
#### create cake 
`/api/cakes/create_cake`
####  view all cakes
 `/api/cakes/available_cakes`
#### view single cake
`/api/cakes/single_cake/<int:id>`
#### edit single cake
`/api/cakes/update_cake/<int:id>`
#### delete_cake
`/api/cakes/delete_cake/<int:id>`

### CODE REVIEW

**Comment by kizzanaome** [comment]
[Fetch a single type of cake and its details.](https://trello.com/c/SWZ6LaRL/41-fetch-a-single-type-of-cake-and-its-details)

**Comment by kizzanaome** [comment]
[Fetch all the types of cakes and their details](https://trello.com/c/JW7Di38I/42-fetch-all-the-types-of-cakes-and-their-details)

**Comment by kizzanaome** [comment]
[Create a cake](https://trello.com/c/263mhkLr/45-create-a-cake)

File: apps/cakes/serializers.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+
+class CakesSerializer(serializers.Serializer):
+    name = serializers.CharField()
+    description = serializers.CharField()
+    price = serializers.IntegerField()
+    rating = serializers.CharField()
+    category = serializers.CharField()
+    tags = serializers.CharField()
+    user_id = serializers.IntegerField(write_only=True)
```
Comment (kdaglas) [inline apps/cakes/serializers.py:13]: what is the use of this id

File: apps/cakes/views.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+@api_view(['GET'])
+def available_cakes(request):
+    all_cakes = Cake.objects.all()
+    serializer = CakesSerializer(all_cakes, many=True)
+    if all_cakes:
+        return Response({"Available_cake":serializer.data})
+    return Response({"message":"There no cakes available"})
+
+@api_view(['POST'])
+def create_cake_view(request):
```
Comment (kdaglas) [inline apps/cakes/views.py:19]: remember that this endpoint is being done at the admin level... a normal user should not be able to do this... that means to do this you need to be querying the DB for a superuser

File: apps/cakes/views.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    cake = request.data
+    serializer = CakesSerializer(data=cake)
+    user_id = request.data['user_id']
```
Comment (kdaglas) [inline apps/cakes/views.py:22]: is this the id for the superuser... what is the DB is cleaned up and then new data starts to be entered... if the first user is a normal user and they carry this id it means they can be able to login to admin... maybe I am not understanding... could you explain a bit

File: apps/cakes/serializers.py (hunk 1 above)
Comment (kizzanaome) [reply apps/cakes/serializers.py:13]: I needed to keep track of the user who created the cake, that's why I thought of adding the user_id to the cake.

File: apps/cakes/views.py (hunk 2 above)
Comment (kizzanaome) [reply apps/cakes/views.py:19]: Thanks for pointing that out @kdaglas am going to rectify it.

File: apps/cakes/views.py (hunk 3 above)
Comment (kizzanaome) [reply apps/cakes/views.py:22]: This id represents the user that has created the cake, I hadn't thought about restricting it to only the admin but let me work on that. @kdaglas

File: apps/cakes/serializers.py (hunk 1 above)
Comment (kdaglas) [reply apps/cakes/serializers.py:13]: ok i get that... cool though in the future you look into making it for admin alone

File: apps/cakes/views.py (hunk 2 above)
Comment (kdaglas) [reply apps/cakes/views.py:19]: ok

File: apps/cakes/views.py (hunk 3 above)
Comment (kdaglas) [reply apps/cakes/views.py:22]: ok

**Pull request merged**